corehq/apps/data_interfaces/tests/test_case_deduplication.py

```diff
@@ -484,99 +560,93 @@ def test_unique_not_updated(self, find_duplicates_mock):
                 .get_case_property('is_potential_duplicate'),
             )
 
-    @patch("corehq.apps.data_interfaces.models.find_duplicate_case_ids")
-    def test_stores_all_duplicates(self, find_duplicates_mock):
-        """When it finds duplicates, store them in the CaseDuplicate model
+    @patch("corehq.apps.data_interfaces.models._find_duplicate_case_ids")
+    def test_stores_new_duplicate(self, find_duplicates_mock):
+        """When it finds a duplicate, store it in the CaseDuplicate model
         """
-        duplicates, uniques = self._create_cases()
+        duplicates, _ = self._create_cases(num_cases=5)
         find_duplicates_mock.return_value = [duplicate.case_id for duplicate in duplicates]
 
         self.rule.run_actions_when_case_matches(duplicates[0])
 
-        # New CaseDuplicate objects should be created for all the duplicate cases
-        self.assertItemsEqual(
-            list(CaseDuplicate.objects.filter(action=self.action).all().values_list('case_id', flat=True)),
-            [duplicate.case_id for duplicate in duplicates]
-        )
+        # We only guarantee that the matching case and at least one other case will get inserted
+        resulting_case_ids = CaseDuplicateNew.objects.filter(
+            action=self.action).all().values_list('case_id', flat=True)
+        self.assertIn(duplicates[0].case_id, resulting_case_ids)
+        self.assertTrue(len(resulting_case_ids) > 1)
```

Comment:
nit: `assertGreater` has richer output if the assertion fails
```suggestion
        self.assertGreater(len(resulting_case_ids), 1)
```

Reply:
Thank you! Was not aware of this assertion